extension/js/common/browser/ui.ts

```diff
@@ -260,6 +262,22 @@ export class Ui {
         }
       });
     },
+    attachmentPreview: async (att: Att, isEncrypted: boolean, factory: XssSafeFactory): Promise<void> => {
+      const iframeUrl = factory.srcPgpAttIframe(att, isEncrypted, 'chrome/elements/attachment_preview.htm');
+      await Ui.swal().fire({
+        onOpen: () => {
+          $(Swal.getContent()).attr('data-test', 'attachment-dialog');
+          $(Swal.getCloseButton()).attr('data-test', 'dialog-close');
+        },
+        html: `<iframe src="${Xss.escape(iframeUrl)}" style="border: 0" sandbox="allow-scripts allow-same-origin"></iframe>`,
```

Comment:
@limonte here, is the allow-scripts necessary? Is it to be able to close the preview with esc or similar functionality?

@seisvelas is there a possibility to use some more restricted method that would not allow inline scripts, but still allow our own extension scripts?

Reply:
> @limonte here, is the allow-scripts necessary? Is it to be able to close the preview with esc or similar functionality?

Yes. `Esc`, handling backdrop clicks to close an attachment.

Reply:
A possible workaround would be to have the "sanboxed" iframe of a limited size (I think we already have), so that the backdrop would still belong to the "outer" frame. After rendering the "sandboxed" frame, set focus on the "outer" frame (is this possible?) so that it can handle the esc event, and also so that it can handle clicks in the backdrop.

It's extra work but if this is possible, then we could run a fully restricted sandbox, with no js and on different origin, which is safer.

Reply:
> @seisvelas is there a possibility to use some more restricted method that would not allow inline scripts, but still allow our own extension scripts?

Not with the sandbox attribute, but we could give it a custom CSP with the `script-src` directive set to `'self'`. 

>It's extra work but if this is possible, then we could run a fully restricted sandbox, with no js and on different origin, which is safer.

Ha, have the unsandboxed iframe clickjack the sandboxed one, kind of. This is definitely the more fun approach. Couldn't we put the parent iframe on top and just give it a transparent body (except for the part we want visible of course), and then put it on top of the child via the z index?

Reply:
> A possible workaround would be to have the "sanboxed" iframe of a limited size (I think we already have), so that the backdrop would still belong to the "outer" frame. After rendering the "sandboxed" frame, set focus on the "outer" frame (is this possible?) so that it can handle the esc event, and also so that it can handle clicks in the backdrop.
> 
> It's extra work but if this is possible, then we could run a fully restricted sandbox, with no js and on different origin, which is safer.

Unfortunately, I don't think that'd be possible without hurting the looks and functionality of the attachment preview. The "outer" frame doesn't know anything about the attachment size (width/height) and that's why it gives all the available space to the "sandboxed" iframe. The "sandboxed" iframe renders the attachment accordingly to its width and height and if there's free space (or backdrop) it'll take care of closing itself by backdrop clicks.

Reply:
> Not with the sandbox attribute, but we could give it a custom CSP with the `script-src` directive set to `'self'`.

Done :+1: